core/trino-main/src/main/java/io/trino/memory/MemoryManagerConfig.java

```diff
@@ -35,10 +35,13 @@
         "resources.reserved-system-memory"})
 public class MemoryManagerConfig
 {
+    public static final String FAULT_TOLERANT_TASK_MEMORY_CONFIG = "fault-tolerant-task-memory";
+
     // enforced against user memory allocations
     private DataSize maxQueryMemory = DataSize.of(20, GIGABYTE);
     // enforced against user + system memory allocations (default is maxQueryMemory * 2)
     private DataSize maxQueryTotalMemory;
+    private DataSize faultTolerantTaskMemory = DataSize.of(1, GIGABYTE);
```

Comment:
The target input size is set to 1GB. I wonder if it is better to set it to something higher, maybe 2 or 3 GB?

Reply:
I think default should be fraction on HEAP, not an absolute value. Will change that in a followup.